mmengine/model/wrappers/test_time_aug.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from abc import abstractmethod
+from typing import Dict, List, Optional, Union
+
+import torch
+
+from mmengine import MODELS
+from mmengine.optim import OptimWrapper
+from mmengine.structures import BaseDataElement
+from ..base_model import BaseDataPreprocessor, BaseModel
+
+# multi-batch inputs processed by different augmentations from the same batch.
+TestTimeAugInputs = List[Union[torch.Tensor, List[torch.Tensor]]]
+# multi-batch data samples processed by different augmentations from the same
+# batch. Each element in `EnhancedDataSamples` means an enhanced batch data.
```

Comment:
Name to EnhancedBatchDataSamples would be better. The outer list stands for different augs and the inner list stands for batch.